include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -86,6 +90,11 @@ class pool_memory_resource final : public device_memory_resource {
       block_from_upstream(initial_pool_size, 0));
   }
 
+  pool_memory_resource(pool_memory_resource const&) = delete;
```

Comment:
Do we need this change for all MRs?

Reply:
It's not necessary because of the `owning_wrapper`. I just did this while debugging. The copy ctor is already being implicitly deleted (because of the custom dtor), so I just made it explicit. 

I think we should be explicitly deleting copy/move operations for resources that we don't want to be copyable/moveable. It's fine for the `cuda/managed_memory_resource` to be copyable/moveable, but probably not any of the sub-allocators. Technically we could make `pool_memory_resource` moveable, but it wasn't really worth the effort at this time since there isn't a need.

Reply:
I'll just do this for all the resources in a separate PR.

Reply:
#445 merged.